File: Study Plan/Top Interview 150/0009 Palindrome Number/PalindromeNumberV2.java
/*  
    Problem
    --------------------------------------------------------------------------------
    9. Palindrome Number
*/

/*  
    Solution
    --------------------------------------------------------------------------------
    - Runtime 4 ms (Beats 100.00% of users with Java)
    - Memory 43.82 MB (Beats 69.62% of users with Java)
*/

public class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0) return false;
        int original = x;
        int reversed = 0;
        while (x > 0) {
            reversed = reversed * 10 + x % 10;
            x /= 10;
        }
        return original == reversed;
    }
}

File: Study Plan/Top Interview 150/0009 Palindrome Number/PalindromeNumberV2.cs
/*  
    Problem
    --------------------------------------------------------------------------------
    9. Palindrome Number
*/

/*  
    Solution
    --------------------------------------------------------------------------------
    - Runtime 43 ms (Beats 80.27% of users with C#)
    - Memory 31.75 MB (Beats 79.16% of users with C#)
*/
public class Solution
{
    public bool IsPalindrome(int x)
    {
        if (x < 0) return false;

        var original = x;
        var reversed = 0;

        while (x > 0)
        {
            reversed = reversed * 10 + x % 10;
            x /= 10;
        }
        
        return original == reversed;
    }
}
